libs/execution_context/ebpf_maps.c

```diff
@@ -861,15 +861,40 @@ ebpf_map_create(
     return result;
 }
 
-uint8_t*
-ebpf_map_find_entry(_In_ ebpf_map_t* map, _In_ const uint8_t* key, int flags)
-{
+ebpf_result_t
+ebpf_map_find_entry(
+    _In_ ebpf_map_t* map,
+    size_t key_size,
+    _In_reads_(key_size) const uint8_t* key,
+    size_t value_size,
+    _Out_writes_(value_size) uint8_t* value,
+    int flags)
+{
+    uint8_t* return_value;
+    if (!(flags & EBPF_MAP_FLAG_HELPER) && (key_size != map->ebpf_map_definition.key_size)) {
+        return EBPF_INVALID_ARGUMENT;
+    }
+
+    if (!(flags & EBPF_MAP_FLAG_HELPER) && (value_size != map->ebpf_map_definition.value_size)) {
```

Comment:
If flag != FLAG_HELPER, i.e. when the caller is UM API, you are enforcing that value size must be equal to definition.value_size? Is that what you wanted?

Reply:
Yes. When we add support for crud operations on per_cpu structures, this will need to be updated to check value_size == size of value in map.

Reply:
Why arent you enforcing that when the call is coming from eBPF program as well? Is it because the verifier does that already?

Reply:
Yes, exactly. Also, eBPF program's don't pass the sizes for keys or values.